pjsip/src/pjsua-lib/pjsua_core.c

```diff
@@ -770,7 +770,7 @@ PJ_DEF(pj_status_t) pjsua_reconfigure_logging(const pjsua_logging_config *cfg)
     pjsua_logging_config_dup(pjsua_var.pool, &pjsua_var.log_cfg, cfg);
 
     /* Redirect log function to ours */
-    pj_log_set_log_func( &log_writer );
+    pj_log_set_log_func((cfg && cfg->cb && !cfg->log_filename.slen) ? cfg->cb : &log_writer);
```

Comment:
As per the discussion in https://github.com/pjsip/pjproject/pull/4086#discussion_r1772695174:
it's better not to change the current behavior.